v4l2/v4l2_decode.cpp

```diff
@@ -39,12 +39,12 @@
 #include "egl/egl_vaapi_image.h"
 
 #define INT64_TO_TIMEVAL(i64, time_val) do {            \
-        time_val.tv_sec = (int32_t)(i64 >> 32);         \
-        time_val.tv_usec = (int32_t)(i64 & 0xffffffff); \
+        time_val.tv_sec = (int32_t)(i64 >> 31);         \
+        time_val.tv_usec = (int32_t)(i64 & 0x7fffffff); \
     } while (0)
 #define TIMEVAL_TO_INT64(i64, time_val) do {            \
         i64 = time_val.tv_sec;                          \
-        i64 = (i64 << 32) + time_val.tv_usec;           \
+        i64 = (i64 << 31)  + time_val.tv_usec;  \
```

Comment:
time_val is fake here.
timestamp is usually int64_t from demuxer, but v4l2 uses time_val. so I simply split int64 to two int32 of time_val.
when we feed the lowest 32bit of int64 to a int32, it may overflow int32. so I just feed the lowest 31 bit to the first int32; the reset bits to the second int32. 
(potential issue is that valid bits in int64 is 62 bits, not 63bits as before.  however, it's big enough; and much bigger than the capacity of time_val)